src/core/cgroup.c

```diff
@@ -2959,6 +2959,9 @@ static int on_cgroup_empty_event(sd_event_source *s, void *userdata) {
                         log_debug_errno(r, "Failed to reenable cgroup empty event source, ignoring: %m");
         }
 
+        /* Update state based on systemd-oomd kills before we notify about cgroup empty event */
+        (void) unit_check_oomd_kill(u);
```

Comment:
this should probably be a separate commit, no? and we should probably do unit_check_oom() here too, just in case.

Reply:
(i.e. as in manager_dispatch_sigchld())